src/script/lua_api/l_client.cpp

```diff
@@ -20,14 +20,38 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 
 #include "l_client.h"
 #include "l_internal.h"
+#include "util/string.h"
 
 int ModApiClient::l_get_current_modname(lua_State *L)
 {
 	lua_rawgeti(L, LUA_REGISTRYINDEX, CUSTOM_RIDX_CURRENT_MOD_NAME);
 	return 1;
 }
 
+
+// send_chat_message()
+int ModApiClient::l_send_chat_message(lua_State *L)
```

Comment:
i agree with say but i don't like mods to send chat messages to server without user agreement. 

Cons:

* Message flooding using Lua API
* Sending messages without user agreement
* Craft bogus messages easily which can annoy other users experience